packages/common/core-interfaces/src/test/jsonStringifyAndParse.spec.ts

```diff
@@ -0,0 +1,2701 @@
+						setOfNumbers instanceof Set,
+						"mapOfStringsToNumbers is an instance of Set",
+					);
+					assert.ok(!(out instanceof Set), "out is not an instance of Set");
+				});
+			});
+
+			describe("Fluid types", () => {
+				it("`IFluidHandle`", () => {
+					const { stringified, out } = stringifyThenParse(
+						// @ts-expect-error SerializationErrorPerNonPublicProperties
+						fluidHandleToNumber,
+						{ isAttached: false },
+					);
+					assertIdenticalTypes(
+						stringified,
+						createInstanceOf<JsonString<typeof fluidHandleToNumber>>(),
+					);
+					assertIdenticalTypes(out, createInstanceOf<{ readonly isAttached: boolean }>());
+				});
+				it("object with `IFluidHandle`", () => {
+					const { stringified, out } = stringifyThenParse(
+						// @ts-expect-error SerializationErrorPerNonPublicProperties
+						objectWithFluidHandle,
+						{ handle: { isAttached: false } },
+					);
+					assertIdenticalTypes(
+						stringified,
+						createInstanceOf<JsonString<typeof objectWithFluidHandle>>(),
+					);
+					assertIdenticalTypes(
+						out,
+						createInstanceOf<{ handle: { readonly isAttached: boolean } }>(),
+					);
+				});
+			});
+		});
+	});
+
+	describe("special cases", () => {
+		it("explicit `any` generic still limits allowed types", () => {
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+			const stringified = JsonStringify<any>(
+				// @ts-expect-error `any` is not an open door (expects `JsonTypeWith<never> | OpaqueJsonSerializable<unknown>`)
+				undefined,
+			);
+			assert.strictEqual(stringified, undefined);
+		});
+
+		describe("`number` edge cases", () => {
+			describe("supported", () => {
+				it("MIN_SAFE_INTEGER", () => {
+					const { stringified, out } = stringifyThenParse(Number.MIN_SAFE_INTEGER);
+					assertIdenticalTypes(stringified, createInstanceOf<JsonString<number>>());
+					assertIdenticalTypes(out, createInstanceOf<number>());
+				});
+				it("MAX_SAFE_INTEGER", () => {
+					const { stringified, out } = stringifyThenParse(Number.MAX_SAFE_INTEGER);
+					assertIdenticalTypes(stringified, createInstanceOf<JsonString<number>>());
+					assertIdenticalTypes(out, createInstanceOf<number>());
+				});
+				it("MIN_VALUE", () => {
+					const { stringified, out } = stringifyThenParse(Number.MIN_VALUE);
+					assertIdenticalTypes(stringified, createInstanceOf<JsonString<number>>());
+					assertIdenticalTypes(out, createInstanceOf<number>());
+				});
+				it("MAX_VALUE", () => {
+					const { stringified, out } = stringifyThenParse(Number.MAX_VALUE);
+					assertIdenticalTypes(stringified, createInstanceOf<JsonString<number>>());
+					assertIdenticalTypes(out, createInstanceOf<number>());
+				});
+			});
+			describe("resulting in `null`", () => {
+				it("NaN", () => {
+					const { stringified, out } = stringifyThenParse(Number.NaN, null);
+					assertIdenticalTypes(stringified, createInstanceOf<JsonString<number>>());
+					// However, real result is `null`
```

Comment:
Want to actually assert this?

```suggestion
					assert.equal(out, null, "However, real result is `null`");
```